pkg/diff/diff.go

```diff
@@ -431,23 +566,78 @@ func (t *TableDiff) compareRows(sourceRows map[string][]map[string]*dbutil.Colum
 }
 
 // WriteSqls write sqls to file
-func (t *TableDiff) WriteSqls(ctx context.Context, writeFixSQL func(string) error) {
-	for {
-		select {
-		case dml, ok := <-t.sqlCh:
-			if !ok || dml == "end" {
+func (t *TableDiff) WriteSqls(ctx context.Context, writeFixSQL func(string) error) chan bool {
+	t.wg.Add(1)
+	stopWriteCh := make(chan bool)
+
+	go func() {
+		defer t.wg.Done()
+
+		stop := false
+		for {
+			select {
+			case dml, ok := <-t.sqlCh:
+				if !ok {
+					return
+				}
+
+				err := writeFixSQL(fmt.Sprintf("%s\n", dml))
+				if err != nil {
+					log.Error("write sql failed", zap.String("sql", dml), zap.Error(err))
+				}
+				t.wg.Done()
+			case <-stopWriteCh:
+				stop = true
```

Comment:
only mark `stop = true` rather than `return`, so we can get all dml from `t.sqlCh`?

Reply:
yes, if can read data from t.sqlCh, will not run default